### PR TITLE
[DISCUSS] QImg intial commit

### PR DESCRIPTION
**Pull request by BenoitRanque**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch and _not_ the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] It's been tested with all Quasar themes
- [ ] It's been tested on a Cordova (iOS, Android) app
- [ ] It's been tested on a Electron app
- [ ] Any necessary documentation has been added or updated [in the docs](https://github.com/quasarframework/quasar-framework.org/tree/master/source) (for faster update click on "Suggest an edit on GitHub" at bottom of page) or explained in the PR's description.

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

This component allows lazy load of images. Please discuss and view comments to understand inner workings, I am open to changing things if you have any suggestions. Once given green light, will write docs & remove unnecessary comments

### CODE REVIEW

File: src/components/img/QImg.vue [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        :src="placeholder"
+        class="preview"
+        :alt="$attrs.alt"
+      >
+    </transition>
+    <transition name="reveal" @after-enter="state += 1">
+      <img
+        v-if="state > 0"
+        v-show="state > 1"
+        @load="state += 1"
```
Comment (dennythecoder) [inline src/components/img/QImg.vue]: Is this supposed to be capturing the native onload event?

Comment (BenoitRanque) [reply src/components/img/QImg.vue]: Yes. Reference [here](http://jsfiddle.net/nobu222/xr0g3Lco/)

Comment (dennythecoder) [reply src/components/img/QImg.vue]: Neat, I don't think I would have thought to have done that. If this event handlers were changed to:
`@after-enter="state = 3"`
`@load="state = 2"`
That'd be accurate, right?  And more explicit?

Comment (BenoitRanque) [reply src/components/img/QImg.vue]: Done

**Comment by pdanpdan** [comment]
Check these ones too:
https://github.com/hilongjw/vue-lazyload
https://github.com/alexjoverm/v-lazy-image

The second one seems light enough to include.

**Comment by BenoitRanque** [comment]
I like the second one. Note the use of the [intersection observer api](https://caniuse.com/#feat=intersectionobserver)

**Pull request closed without merging**